tests/integration/dhcp_test.py

```diff
@@ -129,6 +134,11 @@ def dhcp_env():
         _clean_up()
 
 
+@pytest.fixture(scope='function', autouse=True)
+def reset_tmo():
```

Comment:
Not sure what this is for exactly, but I would avoid using globals for the timeout values.
Either embed them in the retry func or ask to input it explicitly (and use the default argument if you want).